packages/components/src/toggle-group-control/toggle-group-control-option-base/component.tsx

```diff
@@ -52,7 +52,9 @@ function ToggleGroupControlOptionBase(
 			false
 		>,
 		// the element's id is generated internally
-		'id'
+		| 'id'
+		// due to how the component works, only the `disabled` prop should be used
+		| 'aria-disabled'
```

Comment:
I don't think it makes sense to have the options `accessibleWhenDisabled` cc @mirka

Reply:
I actually thought this was one of the cases where we _always_ want the option to be focusable when disabled, similar to ToolbarButton (#63102) or tabs in a tablist. It's also a [composite widget](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#focusabilityofdisabledcontrols) where there will be a limited number of options, and the number of tab stops it potentially adds to the tab sequence is either zero or negligible.

Implementing that in the radio case for ToggleGroupControl would require some additional design work though 🤔 But at least Ariakit supports the behavior nicely ([StackBlitz demo](https://stackblitz.com/edit/qw4xvq?file=radio%2Findex.tsx)).

Reply:
> Implementing that in the radio case for ToggleGroupControl would require some additional design work though

Yep. Currently, since `disabled` options are not really supported, the backdrop indicator also works as a focus indicator. 
If we make the options `accessibleWhenDisabled`, then we'll need to add separate focus rings.

> But at least Ariakit supports the behavior nicely ([StackBlitz demo](https://stackblitz.com/edit/qw4xvq?file=radio%2Findex.tsx)).

Yup. We'll probably need to add the custom logic when rendering individual `button`s